### DISCUSSION TITLE
Bot gets stuck in the shuttle

### DISCUSSION THREAD

**Discussion by pavel4win** [FakeFishGames/Barotrauma#13786] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [X] My issue happened while using mods.

### What happened?

Bot gets stuck in the "Cannonball" shuttle

The bot freely enters the shuttle, but cannot exit.
If I call him, he stomps around inside the shuttle at the exit, saying “I can’t get through.”
Even if in a diving suit he can not pass.

This happens in the multiplayer campaign.
When I loaded this sub into the editor and ran the "test"
This error did not appear.

I noticed that before this an error appears in the console:
"Creating hulls between docking ports failed. Could not find a hull next to the docking port."
However, this error is probably unimportant and applies to the second shuttle "rusting steel"
https://steamcommunity.com/sharedfiles/filedetails/?id=3014307661

Perhaps this is not a game error, but a submarine error?
I am sending you the save of the multiplayer game.

The submarine is 
https://steamcommunity.com/sharedfiles/filedetails/?id=2927095567
I slightly modified the sub in the editor
The error appeared the same before and after modification.
Is there a guide to creating a proper docking port for the shuttle?

[glassCannon.zip](https://github.com/FakeFishGames/Barotrauma/files/15012014/glassCannon.zip)

### Reproduction steps

1. start server
2. unpack and load save
3. order bot "to follow"
4. enter the shuttle in and out

### Bug prevalence

Happens regularly

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.3.0.4

### Which operating system did you encounter this bug on?

Windows

**Comment by SomeRandomNoobKekeke** [comment]
I think these waypoints are a mess
![20240418143212_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/939215dd-a732-445f-8b70-d08dff295dbf)
![20240418143218_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/01755a65-9555-44ec-8cbe-27fa5eea04e3)
![20240418152831_1](https://github.com/FakeFishGames/Barotrauma/assets/122838333/5e34e0eb-ee10-4a70-aebe-0c7b79d7365a)

**Comment by pavel4win** [comment]
Thank for your opinion.
I don't have any experience in placing waypoints, please, can you advise how I can learn how to do this. Is there a guide on how to place waypoints? Especially for submarines with a shuttle, that is, with variable geometry.

Please advise how you managed to get the submarine and the shuttle at the same time in the editor?
I can only load and edit the submarine and the shuttle separately.

Or is it not the edit mode shown in the screenshots?

**Reply by SomeRandomNoobKekeke** [reply]
[That's all we have ](https://regalis11.github.io/BaroModDoc/Editors/SubmarineEditor.html#waypoints) 
![barodev small](https://github.com/FakeFishGames/Barotrauma/assets/122838333/500fa7e0-4828-4bd6-bca7-f139f896fadd)

I can say that if you have such doors with separated buttons then you need waypoints on each button or bot's not gonna press them
And you want to keep those lines smooth and non-intersecting
Actually to close or too far waypoints could be a problem
Or bots can just stuck for no reason because they're bots

You should find some custom subs and see how it's done there

**Reply by pavel4win** [reply]
Thanks, I'll look at the waypoints on the Remora sub.

**Comment by pavel4win** [comment]
How you managed to get the submarine and the shuttle at the same time in the editor?
I can only load and edit the submarine and the shuttle separately.

Or is it not the edit mode shown in the screenshots?

**Reply by SomeRandomNoobKekeke** [reply]
I just run test, undocked, typed debugai and took a shot
Yes, idk why he putted drone in same sub, very inconvenient

**Comment by pavel4win** [comment]
Thanks for your help. I fixed the points and the bots now work fine.
